README.md

```diff
@@ -19,12 +19,351 @@ These dependencies need to be installed, in addition to ROS:
+The following example shows how to create the strategy for the `Camera3dRecordingDesire` previously defined.
+```cpp
+// The template argument of the class declares the desire type associated with the strategy.
+auto strategy = make_unique<Strategy<Camera3dRecordingDesire>>(
+        utility, // The utility of the strategy
+        unordered_map<string, uint16_t>{}, // Declare the resources used by the strategy
+        unordered_map<string, FilterConfiguration>{
+            {"video_recorder_camera_3d/filter_state", FilterConfiguration::onOff()}}, // Declare the filters to enable and their type.
+        move(filterPool)); // The instance of the class to change the filter state.
+```
+Generally, the previous code is put inside a function to simplify its use in other projects.
+The association between the desire and the strategy is done with the template argument of the `Strategy` class.
+
+The following example shows how to create a subclass in order to publish a message when the strategy is enabled.
+Also, the strategy removes the desire of the set once the desire is completed.
+```cpp
+// The template argument of the Strategy class declares the desire type associated with the strategy.
+class TalkStrategy : public Strategy<TalkDesire>
+{
+    std::shared_ptr<DesireSet> m_desireSet;
+    ros::NodeHandle& m_nodeHandle;
+    ros::Publisher m_talkPublisher;
+    ros::Subscriber m_talkDoneSubscriber;
+
+public:
+    TalkStrategy(
+        uint16_t utility, // The utility of the strategy
+        std::shared_ptr<FilterPool> filterPool, // The instance of the class to change the filter state.
+        std::shared_ptr<DesireSet> desireSet,
+        ros::NodeHandle& nodeHandle)
+        : Strategy<TalkDesire>(
+            utility,
+            {{"sound", 1}}, // Declare the resources used by the strategy.
+            {{"talk/filter_state", FilterConfiguration::onOff()}}, // Declare the filters to enable and their type.
+            move(filterPool)),
+        m_desireSet(move(desireSet)),
+        m_nodeHandle(nodeHandle)
+    {
+        // Create the publisher to send the text to say.
+        m_talkPublisher = nodeHandle.advertise<talk::Text>("talk/text", 1, true);
+
+        // Create the subscriber to be notified when the text has been said.
+        m_talkDoneSubscriber = nodeHandle.subscribe("talk/done", 10, &TalkStrategy::talkDoneSubscriberCallback, this);
+    }
+
+    DECLARE_NOT_COPYABLE(TalkStrategy);
+    DECLARE_NOT_MOVABLE(TalkStrategy);
+
+protected:
+    // Override the method to publish a message when the strategy is enabled.
+    void onEnabling(const TalkDesire& desire) override
+    {
+        Strategy<TalkDesire>::onEnabling(desire); // Enable the filters declared in the constructor.
+
+        // Publish the text to be said.
+        talk::Text msg;
+        msg.text = desire.text();
+        msg.id = desire.id();
+        m_talkPublisher.publish(msg);
+    }
```

Comment:
Maybe use a [Template Method](https://refactoring.guru/design-patterns/template-method) here?
I think that the necessity to call the superclass `onEnabling` here to enable the filters is a risk of bugs.

Reply:
It was already the case, but I did not add the call to the BaseStrategy::onEnabling in case you did not want to enable the filters declared in the constructor. This is not really useful, so I removed the feature.